src/frame/mod.rs

```diff
@@ -251,6 +265,18 @@ impl From<Result<Response, ExceptionResponse>> for ResponsePdu {
     }
 }
 
+impl<T> From<Option<T>> for OptionalResponsePdu
+where
+    T: Into<ResponsePdu>,
+{
+    fn from(from: Option<T>) -> OptionalResponsePdu {
```

Comment:
- Use `Self` instead of concrete types whenever possible
- Couldn't this expression be simplified? `Self(from.map(Into::into))`